types/index.d.ts

```diff
@@ -1187,7 +1187,7 @@ declare namespace RamdaAdjunct {
          * Creates an array with all falsy values removed.
          * The values false, null, 0, "", undefined, and NaN are falsy.
          */
-        compact<T>(list: T[]): Array<NonNullable<T>>;
+        compact<T>(list: T[]): Array<Exclude<NonNullable<T>, false | '' | 0>>;
```

Comment:
Aren't we missing `NaN`. `NaN` is not part of `NonNullable`.

Reply:
Unfortunately I'm pretty sure that they type of `NaN` is just `number` and that there is no more specific way to match it.

Like right now in the typescript APIs it just has the following to define NaN:
```typescript
declare var NaN: number;
```

So it doesn't have its own type. There seems to be an open proposal to add it:
https://github.com/Microsoft/TypeScript/issues/28682

Reply:
Cool thank for explanation.